### PR TITLE
prometheus handler

### PR DESCRIPTION
**Pull request by dougfort**
Initial request to get CircleCI check

### CODE REVIEW

**Comment by dougfort** [comment]
We capture
```go
//      http_request_duration_seconds
//      http_request_size_bytes
//      http_response_size_bytes
```

**Comment by dougfort** [comment]
See https://prometheus.io/docs/practices/histograms/#quantiles

```
To calculate the average request duration during the last 5 minutes from a histogram or summary called http_request_duration_seconds, use the following expression:

rate(http_request_duration_seconds_sum[5m]) / rate(http_request_duration_seconds_count[5m])
<img width="1391" alt="screen shot 2018-06-08 at 1 09 00 pm" src="https://user-images.githubusercontent.com/17745/41171073-323f49c8-6b1d-11e8-84b2-ee254f184a0a.png">

```

**Comment by dougfort** [comment]
See https://prometheus.io/docs/practices/histograms/#quantiles
```To calculate the average request duration during the last 5 minutes from a histogram or summary called http_request_duration_seconds, use the following expression:

  rate(http_request_duration_seconds_sum[5m]) / rate(http_request_duration_seconds_count[5m])
```

**Comment by dougfort** [comment]
screen shot 2018-06-08 at 1 09 00 pm

**Comment by dougfort** [comment]
See: https://povilasv.me/prometheus-tracking-request-duration/

```
Also we could calculate percentiles from it. Prometheus comes with a handy histogram_quantile function for it. For example calculating 50% percentile (second quartile) for last 10 minutes in PromQL would be:

histogram_quantile(0.5, rate(http_request_duration_seconds_bucket[10m]))
```

**Comment by dougfort** [comment]
screen shot 2018-06-08 at 1 14 44 pm

File: metrics/prometheus/http_wrapper.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	"github.com/deciphernow/gm-fabric-go/metrics/httpmetrics"
+)
+
+var (
+	// LabelNames is the list of valid lable names for this metric
+	LabelNames = []string{"key", "method", "status"}
+
+	// DefaultKeyFunc returns the URI as the metrics key
+	DefaultKeyFunc KeyFunc = func(req *http.Request) string {
+		return req.URL.EscapedPath()
```
Comment (tilleryd) [inline metrics/prometheus/http_wrapper.go]: @dougfort Using the URL path in a label can lead to high cardinality labels and potentially overload Prometheus.

Comment (dougfort) [reply metrics/prometheus/http_wrapper.go]: Yes. @joshua-rutherford brought that up too. We (or I) need to develop a key function that creates better labels. Note that @rfielding has a similar issue in the metrics for gm-data where there are thousands of URIs with file names.

**Pull request merged**